ctapipe/image/extractor.py

```diff
@@ -987,36 +991,23 @@ def _apply_second_pass(
             keep_isolated_pixels=False,
             min_number_picture_neighbors=1,
         )
-        image_1 = charge_1stpass.copy()
-        image_1[~mask_1] = 0
 
         # STEP 3
 
-        # find all islands using this cleaning
-        num_islands, labels = number_of_islands(camera_geometry, mask_1)
```

Comment:
Why was this removed? Changing how this code handles islands is not just fixing the bug you mentioned in the title.

Reply:
Yes I just noticed this - without this the algorithm behaves differently than the original MARS one (which is the point - if we want we can create variations of course, but the 2nd pass time gradient is supposed to use the main cluster)

Reply:
It's not removed - It's just moved a few lines below where it is actually used.

Reply:
ah, ok, sorry.

Reply:
No, it's still different, because now the check on the number of pixels and the parameterization are performed on the multi-cluster image.

Whar instead this algorithm is supposed to do is to clean for the biggest cluster, check if n_pixels > 3, if yes parametrized the image, and then perform the timing fit.

I report here the explanation of Abelardo,

> [...] After this cleaning, only the biggest cluster of pixels ("main island") is kept. If the resulting image has three or more pixels, it is parametrized following the Hillas approach. A linear fit of pulse time vs. distance along major image axis is then performed [...]

Reply:
Ah, I see now - it was very confusing to follow with all the copies of the images, and each one being overwritten.  I'll modify it to check the number of pixels in the "biggest island" mask.